### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by paiwandkarem**
Student ID: s4537069 
Task: DCGAN for the OASIS Brain Dataset

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- *Submitted after deadline*
- Code seems reasonable for Hard Difficulty
- Source appropriately referenced and credited
- Results seem good
*Other Feedback*: could use more comments for loss, activation and training loop. It would be good to seperate the code for building the model from the main file.

**Comment by rexsw** [comment]
Code looks good for hard difficulty

For next time;

Code in files like driver scripts should be in a main function and in a python file.

Paths such as 'C:\\Users\\paiwa\\Desktop\\Uni\\Sem 2 2020\\COMP3710\\keras_png_slices_data\\keras_png_slices_test\\*.png\
" won’t work for the user and should either be set dynamical or it needs to be made clear to the user that they need to change these paths

Comments such as 
   # Pre processing and image displaying functions\
    def pre_processing(input):
Add nothing to the readability of the code and should be avoided. Overall comments were well done

**Comment by paiwandkarem** [comment]
> Code looks good for hard difficulty
> 
> For next time;
> 
> Code in files like driver scripts should be in a main function and in a python file.
> 
> Paths such as 'C:\Users\paiwa\Desktop\Uni\Sem 2 2020\COMP3710\keras_png_slices_data\keras_png_slices_test\*.png
> " won’t work for the user and should either be set dynamical or it needs to be made clear to the user that they need to change these paths
> 
> Comments such as
> # Pre processing and image displaying functions
> def pre_processing(input):
> Add nothing to the readability of the code and should be avoided. Overall comments were well done

Inside my the Run Model folder I actually have a driver file in its on main function along with dynamic paths. I thought I'd just respond here in case this is something that can affect my marks.

File: recognition/s4537069-DCGAN-OASIS/Run Model/model.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
+import tensorflow as tf
```
Comment (shakes76) [inline recognition/s4537069-DCGAN-OASIS/Run Model/model.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. remove unnecessary epoch image files.**
**Add more comments to driver. Notebooks have markdown sections for you to explain what's going on, so you should make use of it.**
No Proper Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Comment by shakes76** [comment]
Found the assignment extension approval from school. Will apply to marks.

**Comment by paiwandkarem** [comment]
Hi, I have updated per the feedback comments.
